### PR TITLE
Basic profile photo upload functionality added.

### PR DESCRIPTION
**Pull request by saadyousafarbi**
User will now be able to do the following:

* Upload his/her `profile_photo` on the `profile/edit/` page,
* View his/her `profile_photo` on the `profile` page,

`AJAX` would be used to edit profile and send `profile_photo` in request with the help of `JQuery`

### CODE REVIEW

File: core/models.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Profile(models.Model):
     gender = models.CharField(
         max_length=10, help_text='Specify gender.', choices=GENDER_CHOICES, null=True
     )
-    photo = models.ImageField(help_text='Profile picture.', null=True)
+    photo = models.ImageField(help_text='Profile picture.', null=True, upload_to='photo/')
```
Comment (zubair-arbi) [inline core/models.py]: I don't think we need to specify the upload folder. Let the django decide.

File: core/views.py [hunk 2]
```diff
@@ -135,26 +136,28 @@ def edit_profile(request):
...
+    #
+    #         Profile.objects.filter(user=request.user).update(
+    #             bio=form.data['bio'],
+    #             gender=form.data['gender'],
+    #             photo=request.FILES['profile_photo'],
+    #             mobile_number=form.data['mobile_number'],
+    #             address=form.data['address'],
+    #             city=form.data['city'],
+    #             country=form.data['country'],
+    #         )
```
Comment (zubair-arbi) [inline core/views.py]: Never commit commented code.

**Review by zubair-arbi — CHANGES_REQUESTED** [review]

File: core/forms.py [hunk 3]
```diff
@@ -54,6 +54,8 @@ class EditProfileForm(ModelForm):
     """
     Edit Profile form.
     """
+    profile_photo = forms.ImageField(help_text='Upload profile photo', required=False)
```
Comment (zubair-arbi) [inline core/forms.py]: Why not use the model field `photo`?

File: analytics/migrations/0002_auto_20170828_1105.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    dependencies = [
+        ('analytics', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='analyticsrequestsrecord',
+            name='date',
+            field=models.DateField(default=datetime.date.today, unique=True, verbose_name=b'Record Date'),
```
Comment (zubair-arbi) [inline analytics/migrations/0002_auto_20170828_1105.py]: why there is a migration when there is no modification in `analytics` model in this PR?

File: core/migrations/0002_auto_20170828_1105.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('core', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='profile',
+            name='photo',
+            field=models.ImageField(help_text=b'Profile picture.', null=True, upload_to=b'photo/'),
```
Comment (zubair-arbi) [inline core/migrations/0002_auto_20170828_1105.py]: delete this migrations as its no longer needed

File: core/views.py [hunk 6]
```diff
@@ -140,9 +141,11 @@ def edit_profile(request):
         post_data['date_of_birth'] = parse(post_data['date_of_birth'])
         form = EditProfileForm(post_data, instance=user_profile)
         if form.is_valid():
+
```
Comment (zubair-arbi) [inline core/views.py]: remove this extra newline

File: core/views.py [hunk 7]
```diff
@@ -5,6 +5,7 @@
 from django.contrib.auth import authenticate, login, logout
 from django.contrib.auth.decorators import login_required
 from django.contrib.auth.models import User
+from django.core.files.storage import FileSystemStorage
```
Comment (zubair-arbi) [inline core/views.py]: remove this import as its not being used

**Review by zubair-arbi — CHANGES_REQUESTED** [review]

**Comment by saadyousafarbi** [comment]
I accidentally deleted one of the review comments but have made the changes.

File: core/forms.py (hunk 3 above)
Comment (zubair-arbi) [inline core/forms.py:57]: why not use field `photo` of model `Profile`

Comment (saadyousafarbi) [reply core/forms.py:57]: there was a problem with using `ModelForm` field of `photo` from `profile` model. The `photo` field was always empty and the working solution to that was to introduce a separate form field.

**Review by zubair-arbi — APPROVED** [review]
lgtm 👍

**Comment by saadyousafarbi** [comment]
squashed from 719db5c40553d1d3a0e88490e87bb517abc8b2a7

**Pull request merged**